### DISCUSSION TITLE
How to use HttpExporterBuilder?

### DISCUSSION THREAD

**Discussion by l1x** [open-telemetry/opentelemetry-rust#1465] [General]
Right now this is only way I can compile opentelemetry-http:

```Rust
    let client = reqwest::Client::new();
    let exporter: Result<LogExporter, LogError> = opentelemetry_otlp::new_exporter()
        .http()
        .with_export_config(export_config)
        .with_headers(auth)
        .with_http_client(client)
        .build_log_exporter();
```

This means I need to add reqwest = "0.11" as a dependency to my project and I am not sure why we have opentelemetry-http = { version = "0.10", features = ["reqwest"] }. Is there a way to use request as a feature of  opentelemetry-http?

**Comment by TommyCpp** [comment]
opentelemetry-http defines a trait to abstract difference between HTTP client implementations. To help users we also provided implementation for popular http client options. Thus, the `reqwest` feature doesn't contains the reqwest client but it rather provides the http trait implementation for reqwest in opentelemetry-http

**Reply by l1x** [reply]
Thanks!
